src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea():number;
 }
 
-export class Triangle {
+abstract class BasedForm implements Figure {
+  abstract shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
+
+  protected checkAllSides(sides: number[]): void {
+    if (sides.some((value: number) => value <= 0)) {
+      throw new Error(
+        `figure of shape ${this.shape} should have only positive params`,
+      );
+    }
+  }
 }
 
-export class Circle {
+export class Triangle extends BasedForm {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    this.checkAllSides([a, b, c]);
+
+    const sortedSides: number[] = [a, b, c]
+      .sort((aSide: number, bSide: number) => bSide - aSide);
 
+    if (sortedSides[0] >= sortedSides[1] + sortedSides[2]) {
+      throw new Error('some side is uncorrect');
+    }
+  }
+
+  getArea(): number {
+    const semiperimetr = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiperimetr
+    * (semiperimetr - this.a)
+    * (semiperimetr - this.b)
+    * (semiperimetr - this.c),
+    );
+
+    return Number(area.toFixed(2));
```

Comment:
Why did you use different rounding logic for different classes? IMO - it's a strange approach, better to create helper function for that purpose